contracts/MigrationContractStaging.cdc

```diff
@@ -0,0 +1,326 @@
+    ///     /storage/self.capsulePathPrefix_ADDRESS_NAME
+    access(all) fun deriveCapsuleStoragePath(contractAddress: Address, contractName: String): StoragePath {
+        return StoragePath(
+                identifier: self.capsulePathPrefix
+                    .concat(self.delimiter)
+                    .concat(contractAddress.toString())
+                    .concat(self.delimiter)
+                    .concat(contractName)
+            ) ?? panic("Could not derive Capsule StoragePath for given address")
+    }
+
+    /* ------------------------------------------------------------------------------------------------------------ */
+    /* ------------------------------------------------ Constructs ------------------------------------------------ */
+    /* ------------------------------------------------------------------------------------------------------------ */
+
+    /********************
+        ContractUpdate
+     ********************/
+
+    /// Represents contract and its corresponding code.
+    ///
+    access(all) struct ContractUpdate {
+        access(all) let address: Address
+        access(all) let name: String
+        access(all) var code: String
+
+        init(address: Address, name: String, code: String) {
+            self.address = address
+            self.name = name
+            self.code = code
+        }
+
+        /// Validates that the named contract exists at the target address.
+        ///
+        access(all) view fun isValid(): Bool {
+            return getAccount(self.address).contracts.names.contains(self.name)
+        }
+
+        /// Serializes the address and name into a string of the form 0xADDRESS.NAME
+        ///
+        access(all) view fun toString(): String {
+            return self.address.toString().concat(".").concat(self.name)
+        }
+
+        /// Returns human-readable string of the Cadence code.
+        ///
+        access(all) view fun codeAsCadence(): String {
+            return String.fromUTF8(self.code.decodeHex()) ?? panic("Problem stringifying code!")
+        }
+
+        /// Replaces the ContractUpdate code with that provided.
+        ///
+        access(contract) fun replaceCode(_ code: String) {
+            self.code = code
+        }
+    }
+
+    /********************
+            Host
+     ********************/
+
+    /// Serves as identification for a caller's address.
+    /// NOTE: Should be saved in storage and access safeguarded as reference grants access to contract staging. If a
+    /// contract host wishes to delegate staging to another account (e.g. multisig account setup enabling a developer
+    /// to stage on its behalf), it should create a PRIVATE Host capability and publish it to the receiving account.
+    ///
+    access(all) resource Host {
+        /// Returns the resource owner's address
+        ///
+        access(all) view fun address(): Address {
+            return self.owner?.address ?? panic("Host is unowned!")
+        }
```

Comment:
Shouldn't this just return `nil` if it can't find the address?

Reply:
On it's own that makes sense. I added a panic here since I'd be panicking in multiple places throughout the contract if nil anyway